### PR TITLE
Add Tox and Travis CI configuration

### PR DESCRIPTION
**Pull request by bd808**
* Add a tox.ini configuration to run tests, flake8, and sphinx.
* Add a .travis.yml configuration to run tox tests
* Fix sphinx issues with module version and docblock parsing
* Fix various minor lint errors identified by flake8

After this lands we can hook up Travis CI notifications for this repo to automatically run tests on PR and merge.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import os
 from setuptools import setup, find_packages
+from mwoauth import __version__
```
Comment (halfak) [inline setup.py]: This breaks basic installation if all dependencies aren't first met before setup begins.  Dependencies can't be checked until setup.py is started.  

```
$pip install mwoauth-0.2.6.tar.gz 
Processing ./projects/mwoauth/dist/mwoauth-0.2.6.tar.gz
    Complete output from command python setup.py egg_info:
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-cmi5s1fe-build/setup.py", line 3, in <module>
        from mwoauth import __version__
      File "/tmp/pip-cmi5s1fe-build/mwoauth/__init__.py", line 3, in <module>
        from .handshaker import Handshaker
      File "/tmp/pip-cmi5s1fe-build/mwoauth/handshaker.py", line 36, in <module>
        from .functions import complete, identify, initiate
      File "/tmp/pip-cmi5s1fe-build/mwoauth/functions.py", line 39, in <module>
        import requests
    ImportError: No module named 'requests'
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /tmp/pip-cmi5s1fe-build/
```

Comment (bd808) [inline setup.py]: Switched to a different technique from https://python-packaging-user-guide.readthedocs.io/en/latest/single_source_version/

File: doc/conf.py [hunk 2]
```diff
@@ -63,9 +65,8 @@ def read(fname):
 # built documents.
 #
 # The short X.Y version.
-version = read("../VERSION").strip()
-# The full version, including alpha/beta/rc tags.
-release = read("../VERSION").strip()
+version = mwoauth.__version__
```
Comment (bd808) [inline doc/conf.py]: Reading directly from the module should work here. Building the docs doesn't have the same chicken and egg problem as running setup.py.

Comment (halfak) [inline doc/conf.py]: +1

File: setup.py [hunk 3]
```diff
@@ -11,9 +11,16 @@ def requirements(fname):
...
+# https://python-packaging-user-guide.readthedocs.io/en/latest/single_source_version/
+base_dir = os.path.dirname(__file__)
+version = {}
+with open(os.path.join(base_dir, 'mwoauth', 'version.py')) as fp:
+    exec(fp.read(), version)
+
 setup(
     name="mwoauth",
-    version="0.2.6",  # Change in mwoauth/__init__.py too
+    version=version['__version__'],
```
Comment (halfak) [inline setup.py]: Nice solution.  I might try that on some other packages.

**Comment by halfak** [comment]
Deployed to pypi in version 0.2.7

Thanks for your work on this!

**Comment by bd808** [comment]
@halfak You're welcome. :)

It would be nice to have this repo setup with https://travis-ci.org/ now so that the tests get run for each pull request and merge.

**Pull request merged**